### DISCUSSION TITLE
[Support]: unable to read exports on HA app on iphone via frigate proxy

### DISCUSSION THREAD

**Discussion by gypais** [blakeblackshear/frigate#12726] [Camera Support]
### Describe the problem you are having

impossible to read exports on the HA application on iphonevia frigate proxy. We see the picture of video but the size is zero and therefore the playback does not start.
It works with an Android phone, strange.

### Steps to reproduce

NA
Version app 2024.7
![WhatsApp Image 2024-08-02 at 15 19 59](https://github.com/user-attachments/assets/ed57c4e7-bba0-45e5-8d80-b1cdbcaa2a09)

### Version

System 0.14.0-7c39b17

### Frigate config file

```yaml
NA
```

### Relevant log output

```shell
NA
```

### Operating system

Other Linux

### Install method

Docker Compose

### Network connection

Wired

### Camera make and model

pg x6 weq 2 + cloudcam

**Comment by NickM-27** [comment]
Have you tried testing without the proxy? This is just loading a file so it doesn't seem like anything that can be fixed on the Frigate side

**Reply by gypais** [reply]
same ! with a chrome browser on iphone.
So it's not a proxy problem !

**Reply by NickM-27** [reply]
That means it's an issue like @hawkeye217 suggested, the iPhone is unable to decode the format that the video is saved in. This ultimately comes back to being a problem of the camera since the data from the camera is saved directly for recordings.

**Reply by gypais** [reply]
I did other tests, I have two cameras.
with the export of one, it's ok with the frigate ip on iphone, but ko with the proxy with app
with the other, it's ko with the frigate ip on iphone and via the proxy with app
So it's more complicated ...
I'll do other tests

**Comment by hawkeye217** [comment]
This could also be caused by using a codec on your camera that isn't supported on iOS. Frigate doesn't do any transcoding on exports.

**Reply by gypais** [reply]
I can read the exported video from a camera if I directly open the frigate URL on the iPhone, but not with my second camera!
With the app, no exported video is readable :(

**Comment by gypais** [comment]
ok, thanks for the info,
What a pain these iphones are, unable to read HEVC H265!
How to transcode the camera stream directly in frigate as soon as the stream is received? before motion analysis or storage even takes place.
Thanks for your help

**Reply by hawkeye217** [reply]
Use go2rtc. 

https://docs.frigate.video/guides/configuring_go2rtc
https://docs.frigate.video/configuration/restream#reduce-connections-to-camera

For example:
```
go2rtc:
  streams:
    CameraName: ffmpeg:rtsp://username:password@192.168.1.123/streamurl#video=h264#hardware#audio=aac

**Reply by gypais** [reply]
I modified the line in go2rtc, like this : 
- "ffmpeg:rtsp://rphn:xxx@192.168.0.243/channel=1_stream=1.sdp#video=h264"
but is that enough? Will frigate use the go2rtc link?
How do I check that the stream has been transcoded?
I don't have to do anything on the cameras block: ?  

my conf : 
go2rtc:
  streams:
    JPP_camera-out:
      #- rtsp://rphn:xxx@192.168.0.243/channel=1_stream=1.sdp
      - "ffmpeg:rtsp://rphn:xxx@192.168.0.243/channel=1_stream=1.sdp#video=h264"

cameras:
  JPP_camera-out:
    detect:
      fps: 5
      enabled: true
    enabled: true
    ffmpeg:
      inputs:
        - path: rtsp://rphn:xxx@192.168.0.243/channel=1_stream=1.sdp
          roles:
            - detect

Thanks.

**Reply by hawkeye217** [reply]
See the docs: https://docs.frigate.video/configuration/restream#reduce-connections-to-camera
